RetailCoder.VBE/Common/Hotkeys/HotkeyFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Rubberduck.Settings;
+using Rubberduck.UI.Command;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Common.Hotkeys
+{
+    public class HotkeyFactory
+    {
+        private readonly IEnumerable<CommandBase> _commands;
+
+        public HotkeyFactory(IEnumerable<CommandBase> commands)
+        {
+            _commands = commands;
+        }
+
+        public Hotkey Create(HotkeySetting setting, IntPtr hWndVbe)
+        {
+            if (setting == null)
+            {
+                return null;
+            }
+
+            var commandToBind = _commands.FirstOrDefault(command => command.GetType().Name == setting.CommandTypeName);
+
+            return commandToBind == null ? null : new Hotkey(hWndVbe, setting.ToString(), commandToBind);
```

Comment:
I think it'd be more readable if it's broken up into:
```
if (commandToBind == null)
{
  return null;
}

return new Hotkey(hWndVbe, setting.ToString(), commandToBind);
```

Reply:
Alternatively, maybe this is a case to implement a null object pattern? A `HotKey` instance with no hotkey to be assigned to those that won't have a hotkey? Eliminate all those null checks.

Reply:
I prefer the ternary operator, to be honest. I hope it's OK if I leave it here. As for the null object pattern, if I return some "empty" hotkey, I will probably run into trouble. In the `Hotkey` class there is an `Attach()` method containing the following code:
```
var key = GetKey(hotKey);

if (key == Keys.None)
{
    throw new InvalidOperationException(Rubberduck.UI.RubberduckUI.CommonHotkey_InvalidKey);
}
```

There's another place in this class where logging takes place if the hotkey can't be successfully registered. I assume an "empty" hotkey would cause writing to this log.

All in all, I think implementing your change would require significant changes to registering hotkeys.

Reply:
By coincidence I just found out that `Hotkey.Attach()` is never used, the same a `Hotkey.Detach()`.